virtual_ecosystem/models/plants/exporter.py

```diff
@@ -0,0 +1,374 @@
+        """A subset of community canopy attribute names to export."""
+        self.stem_canopy_attributes: set[str] = stem_canopy_attributes
+        """A subset of community canopy attribute names to export."""
+
+        if self.active:
+            self._check_attribute_subsets()
+            LOGGER.info("Plant community data exporter active.")
+
+    def _check_paths(self) -> None:
+        """Check paths do not exist and are in existing writeable locations."""
+
+        for arg, fname in (
+            ("cohort_data_path", self.cohort_data_path),
+            ("community_canopy_data_path", self.community_canopy_data_path),
+            ("stem_canopy_data_path", self.stem_canopy_data_path),
+        ):
+            if fname is None:
+                continue
+
+            if fname.exists():
+                msg = f"The {arg} exporter path must not be an existing file: {fname}"
+                LOGGER.error(msg)
+                raise ConfigurationError(msg)
+
+            if not (fname.parent.exists() and os.access(fname.parent, os.W_OK)):
+                msg = (
+                    f"The {arg} exporter path must be in an existing "
+                    f"writeable directory: {fname}"
+                )
+                LOGGER.error(msg)
+                raise ConfigurationError(msg)
+
+    def _check_attribute_subsets(self) -> None:
+        """Check attribute subsets contain available fields."""
+
+        available_attributes = {
+            "cohort_attributes": set(
+                [
+                    "cell_id",
+                    "time",
+                    *StemAllometry.array_attrs,
+                    *Cohorts.array_attrs,
+                    StemAllocation.array_attrs,
+                ]
+            ),
+            "community_canopy_attributes": set(
+                [
+                    "canopy_layer_index",
+                    "heights",
+                    "cell_id",
+                    "time",
+                    *CommunityCanopyData.array_attrs,
+                ]
+            ),
+            "stem_canopy_attributes": set(
+                [
+                    "canopy_layer_index",
+                    "cohort_id",
+                    "cell_id",
+                    "time",
+                    *CohortCanopyData.array_attrs,
+                ]
+            ),
+        }
+
+        for subset_name, available in available_attributes.items():
+            subset = getattr(self, subset_name)
+            # If subset is provided, check the values are all valid
+            if subset:
+                not_found = subset.difference(available)
+                if not_found:
+                    msg = (
+                        f"The {subset_name} exporter configuration contains "
+                        f"unknown attributes: {', '.join(not_found)}"
+                    )
+                    LOGGER.error(msg)
+                    raise ConfigurationError(msg)
```

Comment:
Totally a personal thing, but I feel it is more readable to reduce nesting as much as possible. In this case, I would go for:
```suggestion
            if not subset:
                continue
            
            not_found = subset.difference(available)
            if not_found:
                msg = (
                    f"The {subset_name} exporter configuration contains "
                    f"unknown attributes: {', '.join(not_found)}"
                )
                LOGGER.error(msg)
                raise ConfigurationError(msg)
```
You could even replace the second conditional with:
```python
if not not_found:
    continue
```
But that is probably a step too far and mask a bit that the exceptional circumstance is the error, so I would leave it as it is.